src/interpreter.rs

```diff
@@ -4,12 +4,12 @@ use std::collections::HashMap;
 type Stack<T> = Vec<T>;
 
 #[derive(Default)]
-pub struct Interpreter<'a, T> {
+pub struct Interpreter<'a, 'b, T: 'b> {
```

Comment:
Basically, if you remove `T: 'b` you get the error `error: error: the parameter type 'T' may not live long enough`, and this is a problem because the a `&Word<T>` instance may live longer than the `T` it contains. The compiler directly suggests that fix:
```
error: the parameter type `T` may not live long enough [E0309]
[...]
help: run `rustc --explain E0309` to see a detailed explanation
[...]
help: consider adding an explicit lifetime bound `T: 'b`...
note: ...so that the reference type `&'b interpreter::Word<T>` does not outlive the data it points at
```

Whenever there are multiple references involved, like in function argument or generics, the best way is to let the compiler work for you. For `fn f(x: &T, y:&U) -> &V`, write `fn f<'a,'b,'c>(x: &'a T, y: &'b Y) -> &'V` and let the compiler tell you which lifetime tags are redundant.